core/src/validator.rs

```diff
@@ -1202,6 +1205,22 @@ impl Validator {
             )
             .unwrap();
 
+        let in_wen_restart = config.wen_restart_proto_path.is_some() && !waited_for_supermajority;
+        let tower = match process_blockstore.process_to_create_tower() {
+            Ok(tower) => {
+                info!("Tower state: {:?}", tower);
+                tower
+            }
+            Err(e) => {
+                warn!(
+                    "Unable to retrieve tower: {:?} creating default tower....",
+                    e
+                );
+                Tower::default()
+            }
+        };
+        let last_vote = tower.last_vote();
```

Comment:
Is it sufficient to only look at `last_vote`?
Unlike fork choice `SlotVoteTracker` doesn't subtract stake that moved, so it might have been our `last_last_vote` that contributed to the `67%` threshold.  
Also might be important for heaviest scenarios where the `last_vote` didn't actually land - we think our stake is not on the fork, but other participants viewed our stake as still part of the `last_last_vote` fork.

Reply:
That is exactly why we are sending 81k slots on the last_vote fork, because we need more than last_vote. There are two cases:
- last_last_vote on the same fork as last_vote, then it's on the last_vote fork, every slot on this fork will have stake counted in consensus
- last_last_vote on a different fork than last_vote, in this case consider that the validator continues voting on the new fork, eventually the stakes on last_last_vote's fork will move to the last_vote fork

The wen_restart proposal is trying to send the snapshot of every validator's voting status to everyone and reach consensus, so you can consider your last_vote magically landing everywhere in the restarted cluster, you don't need to consider whether your last_vote landed or not in the pre-restart world, we are starting a new consensus here.

Reply:
> eventually the stakes on last_last_vote's fork will move to the last_vote fork
is this always the case? you could have been an early voter on the optimistic slot, but because of the network slowness, other peoples votes didn't propagate yet. In the meantime you switch to a different fork. Normally this is fine because once the other peoples votes land you'll come back.

just wondering if there's any cases we miss a potentially optimistic slot because `SlotVoteTracker` doesn't track switched stake under the (valid) assumption that any switched stake will return. However `last_vote` relies on the timing of when we enter `wen_restart` which might (?) break this assumption.

Reply:
looked into it more - i'm wrong 😅 .
forgot that unlike duplicate confirmed, OC doesn't roll up replay votes on children, so `last_vote` should be fine.